src/lib/markbind/src/lib/markdown-it-shared/markdown-it-escape-special-tags.js

```diff
@@ -0,0 +1,95 @@
+const markdownIt = require('../markdown-it');
+
+/*
+ Custom patch for the api to escape content in certain special tags
+ Adapted from the default markdown-it html_block rule and replaces it.
+ */
+
+function escape_plugin(md, tagsToIgnore) {
+  const block_names = require('markdown-it/lib/common/html_blocks');
+  const HTML_OPEN_CLOSE_TAG_RE = require('markdown-it/lib/common/html_re').HTML_OPEN_CLOSE_TAG_RE;
+
+  const specialTagsRegex = Array.from(tagsToIgnore)
+    .concat(['script|pre|style'])
+    .join('|');
+  const startingSpecialTagRegex = new RegExp(`^<(${specialTagsRegex})(?=(\\s|>|$))`, 'i');
+  const endingSpecialTagRegex = new RegExp(`<\\/(${specialTagsRegex})>`, 'i');
+
+  const HTML_SEQUENCES = [
+    [ startingSpecialTagRegex, endingSpecialTagRegex, true ],
+    [ /^<!--/,        /-->/,   true ],
+    [ /^<\?/,         /\?>/,   true ],
+    [ /^<![A-Z]/,     />/,     true ],
+    [ /^<!\[CDATA\[/, /\]\]>/, true ],
+    [ new RegExp('^</?(' + block_names.join('|') + ')(?=(\\s|/?>|$))', 'i'), /^$/, true ],
+    [ new RegExp(HTML_OPEN_CLOSE_TAG_RE.source + '\\s*$'),  /^$/, false ]
+  ];
+
+
+  function escape_special_tags(state, startLine, endLine, silent) {
```

Comment:
I also had some difficulty trying to find the specific part of the markdown-it documentation to understand how exactly this interacts with markdown-it. Do you think it would be helpful to link the relevant documentation in a comment?